src/core/Form/TextInput/TextInput.test.tsx

```diff
@@ -257,4 +257,71 @@ describe('props', () => {
       expect(container.contains(icon)).toBeTruthy();
     });
   });
+  describe('visualPlaceholder', () => {
+    it('should have the given text', () => {
+      const { getByTestId } = render(
+        <TextInput
+          labelText="Test input"
+          data-testid="input"
+          visualPlaceholder="Enter text here"
+        />,
+      );
+      const inputField = getByTestId('input') as HTMLInputElement;
+      expect(inputField).toHaveAttribute('placeholder', 'Enter text here');
+    });
+  });
+
+  describe('icon', () => {
+    it('should have the correct classname when icon prop is given', () => {
+      const { container } = render(
+        <TextInput labelText="Test input" icon="close" />,
+      );
+      expect(container.firstChild).toHaveClass('fi-text-input_with-icon');
+    });
+
+    it('should have an icon element when one is specified', () => {
+      const { container } = render(
+        <TextInput labelText="Test input" icon="close" />,
+      );
+      const icon = container.querySelector('.fi-icon');
+      expect(container.contains(icon)).toBeTruthy();
+    });
+  });
+
+  describe('debounce', () => {
+    it('delays the running of onChange by the given time', () => {
+      jest.useFakeTimers();
```

Comment:
This fake timer is great find 👍   
Didn't know it existed before this PR.

Reply:
Thanks. Took some digging to find a way to test this 😄